shared/common-adapters/input.desktop.js

```diff
@@ -84,35 +88,65 @@ class Input extends Component<Props, State> {
       return
     }
 
-    // $FlowIssue
-    node.style.height = '1px'
-    // $FlowIssue
-    node.style.height = `${node.scrollHeight}px`
+    // Try and not style/render thrash
```

Comment:
in order to support growing (and shrinking) as you type we would set the height to 1 and then get the scrollheight and then set it to that height. that causes a ton of redraw in the browser. instead (if the smartAutoresize flag is on) we keep track of the width when we actually grow, when we go smaller than that we remeasure.

Reply:
would be nice to have this somewhere to refer to

Reply:
added more comments